src/Ng/CommandHelpers.cs

```diff
@@ -45,86 +114,113 @@ private static void TraceRequiredArgument(string name)
             Trace.TraceError("Required argument \"{0}\" not provided", name);
         }
 
-        public static string GetSource(IDictionary<string, string> arguments)
+        private static void TraceMissingArgument(string name)
         {
-            string value;
-            if (!arguments.TryGetValue("-source", out value))
+            Console.WriteLine("Argument \"{0}\" not provided", name);
+            Trace.TraceWarning("Argument \"{0}\" not provided", name);
+        }
+
+        private static void TryGetArgument(IDictionary<string, string> arguments, string searchArg, out string value, bool required = false)
+        {
+            if (!arguments.TryGetValue("-" + searchArg, out value))
+            {
+                if (required)
+                {
+                    TraceRequiredArgument(searchArg);
+                    throw new ArgumentException();
+                }
+                else
+                {
+                    TraceMissingArgument(searchArg);
```

Comment:
>TraceMissingArgument [](start = 20, length = 20)

I don't think this is necessary. Not all arguments are required in all scenarios, and printing every missing one just creates noise.